src/mito2/src/read/scan_region.rs

```diff
@@ -779,6 +807,15 @@ impl ScanInput {
         self
     }
 
+    #[cfg(feature = "enterprise")]
+    #[must_use]
+    pub(crate) fn with_extension_ranges(self, extension_ranges: Vec<BoxedExtensionRange>) -> Self {
+        Self {
+            extension_ranges,
+            ..self
+        }
+    }
```

Comment:
We can group methods related to extension ranges in one impl block with the feature gate.
```rust
#[cfg(feature = "enterprise")]
impl ScanInput {
}
```